### PR TITLE
feat: query params

### PR DESCRIPTION
**Pull request by TorstenDittmann**
## What does this PR do?

(Provide a description of what this PR does.)

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/lib/components/pagination.svelte [hunk 1]
```diff
@@ -27,50 +26,62 @@
...
+    function getLink(page: number): string {
+        const url = new URL($pageStore.url);
+        if (page === 1) {
+            url.searchParams.delete('page');
+        } else {
+            url.searchParams.set('page', page.toString());
+        }
+
+        return url.toString();
+    }
```
Comment (TGlide) [inline src/lib/components/pagination.svelte:39]: If we're going to be using searchParams for multiple filters and what not, maybe we could offload this to a helper?

e.g. 
```
function getSearchParamLink(param: string, value: string): string {
    const url = new URL($pageStore.url);
    if (value === '') {
        url.searchParams.delete(param);
    } else {
        url.searchParams.set(param, value);
    }
    return url.toString();
}
```

File: src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/table.svelte [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            {#if column.show}
+                <TableCellHead eyebrow={false}>{column.title}</TableCellHead>
+            {/if}
+        {/each}
+    </TableHeader>
+    <TableBody>
+        {#each data.documents.documents as document}
+            <TableRowLink
+                href={`${base}/console/project-${projectId}/databases/database-${databaseId}/collection-${$collection.$id}/document-${document.$id}`}>
+                <TableCell width={230}>
```
Comment (ArmanNik) [inline src/routes/console/project-[project]/databases/database-[database]/collection-[collection]/table.svelte:75]: Now that the PR that trims this ID has been merged we can reduce the width :)

**Review by ArmanNik — COMMENTED** [review]
Left two small comments. Other than the ID trimming bug that I'm facing the rest LGTM :+1:

File: src/lib/components/searchQuery.svelte [hunk 3]
```diff
@@ -29,23 +29,28 @@
...
+
+            if (value === '') {
+                url.searchParams.delete('search');
+            } else {
+                url.searchParams.set('search', value);
             }
 
-            url.search = value;
             trackEvent('search');
             goto(url, { keepFocus: true });
```
[the remark this answers is not in the document]
Comment (TorstenDittmann) [reply src/lib/components/searchQuery.svelte:55]: makes no difference here :-)

**Review by ArmanNik — APPROVED** [review]

**Pull request merged**